### DISCUSSION TITLE
How should I derive countryCode?

### DISCUSSION THREAD

**Discussion by duttonst** [tidal-music/discussions#87] [Q&A] [answered]
First of all, sorry for all of the questions!

The countryCode query parameter appears to be required on most catalog and playlist queries. Where should this come from?

I had assumed I could get this from the user's profile using `/users/me` but I don't get a country on the response from that endpoint.

Defaulting to US doesn't work either: a playlist created in GB will have track Ids which don't exist in the US catalog.

Thanks in advance!

**Comment by stefanmoro** [comment, marked as answer]
Hey @duttonst, thanks for reaching out.

The countryCode parameter is used by the TIDAL API to correctly present catalogue items for a certain region. It mostly affects availability information about e.g. a track or an album, but it can in some cases also alter other properties of a catalogue entity.

For logged in users, it is recommended to supply the country code where they user is registered, since that is what we validate against for playback. 

You are correct that it should be available in the response from [/users/me](https://developer.tidal.com/apiref?spec=user-v2&ref=get-me-v2). I just checked, and it seems we have not yet exposed that field (even though it is available in the OAS). We will look into fixing that next week. Thank you for pointing this out!

**Reply by duttonst** [reply]
Thank you - that's very helpful. I look forward to seeing this field shortly.

**Reply by stefanmoro** [reply]
A user's count country code should now now be available in the `country` attribute in the response from `users/me`.

Thanks.

**Reply by duttonst** [reply]
Thanks @stefanmoro - I can confirm that's working now.
